src/main/java/frc/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -0,0 +1,264 @@
+package frc.robot.subsystems.swerve;
+
+import com.ctre.phoenix.motorcontrol.*;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.SwerveModuleState;
+import frc.robot.components.TrigonTalonFX;
+import frc.robot.components.TrigonTalonSRX;
+import frc.robot.constants.RobotConstants.SwerveConstants;
+import frc.robot.constants.RobotConstants.SwerveConstants.SwerveModuleConstants;
+import frc.robot.utilities.CTREUtil;
+import frc.robot.utilities.EncoderConversions;
+import frc.robot.utilities.TrigonPIDController;
+
+/**
+ * This class represents a single swerve module.
+ * The module has two motors, one for the drive and one for the angle.
+ * There's a
+ */
+public class SwerveModule {
+    private final double encoderOffset;
+    private final TrigonTalonFX mAngleMotor;
+    private final TrigonTalonFX mDriveMotor;
+    private final TrigonTalonSRX angleEncoder;
+    private final SwerveModuleConstants constants;
+    private final TrigonPIDController driveFeedforward;
+    private SwerveModuleState lastDesiredState;
+
+    public SwerveModule(SwerveModuleConstants moduleConstants) {
+        this.constants = moduleConstants;
+        this.driveFeedforward = new TrigonPIDController(
+                moduleConstants.drivePIDCoefs);
+        encoderOffset = moduleConstants.encoderOffset;
+
+        /* Angle Encoder Config */
+        angleEncoder = new TrigonTalonSRX(moduleConstants.encoderID);
```

Comment:
A why?
B to what?

Reply:
A. let us say you want to use a lift motor as the encoder. instead of playing with ID's and creating two instances of the same motor you do encoder = motor in constants (tho this arises the issue of what do you do with the port from the robot map when you don't use it)